frida/core.py

```diff
@@ -358,11 +417,41 @@ def list_exports(self) -> List[str]:
         assert isinstance(result, list)
         return result
 
+    def list_exports(self) -> List[str]:
+        """
+        List all the exported attributes from the script's rpc
+        """
+
+        warnings.warn(
+            "Script.list_exports will become asynchronous in the future, use the explicit Script.list_exports_sync instead",
+            DeprecationWarning,
+        )
+        return self.list_exports_sync()
+
+    def _rpc_request_async(self, *args: Any) -> asyncio.Future[Any]:
+        loop = asyncio.get_event_loop()
+        future: asyncio.Future[Any] = asyncio.Future()
+
+        def on_complete(value: Any, error: Optional[Union[RPCException, _frida.InvalidOperationError]]) -> None:
```

Comment:
In my opinion this is nicer:
```suggestion
        def on_complete(value: Any, error: Union[None, RPCException, _frida.InvalidOperationError]) -> None:
```

Reply:
Prefer the optional actually, but doesn't really matter